crates/iroha_executor/src/default/isi/multisig/account.rs

```diff
@@ -0,0 +1,64 @@
+//! Validation and execution logic of instructions for multisig accounts
+
+use super::*;
+
+impl VisitExecute for MultisigRegister {
+    fn visit<V: Execute + Visit + ?Sized>(&self, _executor: &mut V) {}
+
+    fn execute<V: Execute + Visit + ?Sized>(self, executor: &mut V) -> Result<(), ValidationFail> {
+        let multisig_account = self.account;
+        let multisig_role = multisig_role_for(&multisig_account);
+
+        // The multisig registrant needs to have sufficient permission to register personal accounts
+        // TODO Loosen to just being one of the signatories? But impose the procedure of propose and approve?
+        visit_seq!(executor
+            .visit_register_account(&Register::account(Account::new(multisig_account.clone()))));
+
+        let domain_owner = executor
+            .host()
+            .query(FindDomains)
+            .filter_with(|domain| domain.id.eq(multisig_account.domain().clone()))
+            .execute_single()
+            .dbg_expect("domain should be found as the preceding account registration succeeded")
+            .owned_by()
+            .clone();
+
+        // Authorize as the domain owner:
+        // Just having permission to register accounts is insufficient to register multisig roles
```

Comment:
why?

Reply:
is it because this account won't have permission to `SetKeyValue`?

Reply:
Because this account can register accounts but may not have permission to register and grant multisig roles